ci/docker/buildenv/Dockerfile

```diff
@@ -8,6 +8,18 @@
 
 FROM ubuntu:22.04
 
+ARG USERNAME=verilator
+ARG USER_UID=1000
+ARG USER_GID=$USER_UID
+
+# Create the user
+RUN groupadd --gid $USER_GID $USERNAME \
+    && useradd --uid $USER_UID --gid $USER_GID -m $USERNAME -s /bin/bash \
+    && apt-get update \
+    && apt-get install -y sudo \
+    && echo $USERNAME ALL=\(root\) NOPASSWD:ALL > /etc/sudoers.d/$USERNAME \
```

Comment:
It is already an improvement, because it was running as root inside the container before. This adds a non-root user and uses this. But if someone starts developing and wants to add packages, they should have sudo rights inside the container. It is more or less a copy from here: https://code.visualstudio.com/remote/advancedcontainers/add-nonroot-user

Beside that its always better to run as non-root user, the approach with the ARG allows devcontainers to mount the current folder mapped between the host user and the container user. It discovers this username and maps UID and GID to the user that started. That way, you can develop inside the container, but still not trash your filesystem with root-owned files.